games-action/dave_gnukem/patches/dave_gnukem-1.0.patchset

```diff
@@ -0,0 +1,179 @@
+ 
+ #ifndef NOSOUND
+-#include <SDL_mixer.h>//For background music stuff
++#include <SDL/SDL_mixer.h>//For background music stuff
+ Mix_Music* g_pGameMusic=NULL;
+ #endif
+ 
+diff --git a/src/graph.cpp b/src/graph.cpp
+index 5d12043..8095da0 100644
+--- a/src/graph.cpp
++++ b/src/graph.cpp
+@@ -14,7 +14,7 @@ Copyright (C) 1998-2018 David Joffe
+ #include <stdio.h>
+ #include <string.h>
+ #include <stdlib.h>
+-#include "SDL.h"
++#include <SDL/SDL.h>
+ #include "sys_log.h"//Log
+ #include "djstring.h"//djStrPrintf
+ 
+diff --git a/src/keys.cpp b/src/keys.cpp
+index abf8fbc..b8abd95 100644
+--- a/src/keys.cpp
++++ b/src/keys.cpp
+@@ -8,7 +8,7 @@ Created: 09/2001
+ 
+ #include "keys.h"
+ #include "settings.h"
+-#include "SDL.h"
++#include <SDL/SDL.h>
+ #include <vector>
+ using namespace std;
+ 
+diff --git a/src/main.cpp b/src/main.cpp
+index 183b95a..a7ef3b9 100644
+--- a/src/main.cpp
++++ b/src/main.cpp
+@@ -51,7 +51,7 @@ Copyright (C) 1995-2018 David Joffe
+ #endif
+ #endif
+ 
+-#include <SDL_mixer.h>//For background music stuff
++#include <SDL/SDL_mixer.h>//For background music stuff
+ 
+ #include <map>
+ #include <string>
+diff --git a/src/sdl/djgraph.cpp b/src/sdl/djgraph.cpp
+index ea61fca..f3a7275 100644
+--- a/src/sdl/djgraph.cpp
++++ b/src/sdl/djgraph.cpp
+@@ -15,7 +15,7 @@ Copyright (C) 1997-2018 David Joffe
+ 
+ #include "../djgraph.h"
+ #include "../sys_log.h"
+-#include "SDL.h"
++#include <SDL/SDL.h>
+ 
+ #include <string.h>
+ 
+diff --git a/src/sdl/djsound.cpp b/src/sdl/djsound.cpp
+index 5a3e2c9..8ebb875 100644
+--- a/src/sdl/djsound.cpp
++++ b/src/sdl/djsound.cpp
+@@ -7,10 +7,10 @@ Copyright (C) 1999-2018 David Joffe and Kent Mein
+ #include "../djsound.h"
+ #include "../djstring.h"
+ #include "../djlog.h"
+-#include <SDL_audio.h>
+-#include <SDL_error.h>
++#include <SDL/SDL_audio.h>
++#include <SDL/SDL_error.h>
+ #ifndef NOSOUND
+-#include <SDL_mixer.h>
++#include <SDL/SDL_mixer.h>
+ #endif
+ 
+ #ifdef __APPLE__
+diff --git a/src/sdl/djtime.cpp b/src/sdl/djtime.cpp
+index 2fa173d..355fe90 100644
+--- a/src/sdl/djtime.cpp
++++ b/src/sdl/djtime.cpp
+@@ -5,7 +5,7 @@ Copyright (C) 1999-2018 David Joffe
+ */
+ 
+ #include "../djtime.h"
+-#include "SDL.h"
++#include <SDL/SDL.h>
+ #ifdef WIN32
+ #include <windows.h>
+ #include <mmsystem.h>
+-- 
+2.19.0
+
+
+From 0140911d8d2d31529531184539d66c08f9dced61 Mon Sep 17 00:00:00 2001
+From: Abdur-Raheem Idowu <abdurraheemidowu@gmail.com>
+Date: Fri, 27 Dec 2019 19:37:24 +0000
+Subject: Allow launch from deskbar
+
+
+diff --git a/src/main.cpp b/src/main.cpp
+index a7ef3b9..e7ad101 100644
+--- a/src/main.cpp
++++ b/src/main.cpp
+@@ -38,6 +38,11 @@ Copyright (C) 1995-2018 David Joffe
+ #include "sys_log.h"
+ #include "sys_error.h"
+ 
++#ifdef __HAIKU__
++#include <libgen.h>
++#include <unistd.h>
++#endif
++
+ #ifdef __APPLE__
+ #include <mach-o/dyld.h>//_NSGetExecutablePath
+ #include <sys/stat.h>//For djFolderExists stuff
+@@ -121,6 +126,10 @@ int main ( int argc, char** argv )
```

Comment:
Started my patch just below https://github.com/davidjoffe/dave_gnukem/blob/abd908d2d8b1c4846affff77259032bc95a23d5f/src/main.cpp#L110
This probably works also, from what I've seen for the other patches it starts the HAIKU ifdef just below the main function @pulkomandy any idea on this?

Reply:
I'll move the patch to just below the main function.